tdrs-backend/docker-compose.ci.yml

```diff
@@ -0,0 +1,52 @@
+# Docker compose file intended to be used in CI/CD pipelines
+version: "3"
+
+services:
+  zaproxy:
+    image: owasp/zap2docker-weekly
+    command: sleep 3600
+  postgres:
+    image: postgres:11.6
+    environment:
+      - POSTGRES_USER=tdpuser
+      - POSTGRES_PASSWORD=something_secure
+      - POSTGRES_DB=tdrs_test
+      - POSTGRES_PORT=5432
+  web:
+    restart: always
+    environment:
+      - DB_USER=tdpuser
+      - DB_PASSWORD=something_secure
```

Comment:
@jtwillis92 Should these be secrets? Sent in through .env?

Reply:
In CI or production contexts these should definitely be environment variables, locally it doesn't matter too much as long as they match with what is passed in to the postgres container.

Reply:
Makes sense -- aren't we looking at `docker-compose.ci.yml` though? 😄

Reply:
Fair point, I'll get that updated to use environment variables for CI :D

Reply:
Great, this is something we could either tackle in this PR or break off as a high-priority follow-on issue. Might be helpful context to note that we currently have a dev site but are planning to add staging and prod next, and these DB variables should be different across each context.

Reply:
Another data point — when I log into Cloud.gov and check out the ENV variables that relate to this database configuration, it looks secure (e.g. it's not "test_user", "password," etc).

So this may not be an active security problem, although it does look that way upon first scan. 

One solution would be to move these hardcoded, plain-text values into just the local environment set up and have something more secure for CI/CD.

Reply:
What I ended up doing was creating three docker-compose files for the backend:
* `docker-compose.yml` stores the base configuration with images and info that doesn't change between environments. In this file the environment has the insecure, hard-coded values.
* `docker-compose.override.yml` - for local dev, the only difference from the base is that this applies the `env_file` directive
* `docker-compose.ci.yml` - for CI/CD, the difference here is that all environment variables passed in to docker via the `environment` directive are extracted from the CI host machine, including the database variables.

These new files follow the docker compose expectation of only defining the differences for each environment and relying on the core configuration from the base compose file.

Reply:
Terrific, sounds like a great approach overall! Although why wouldn't the hard coded values live in `.override.yml`, since that's the local dev one? I believe could co-exist with the `env_file` directive.

A couple of suggestions:

* If you could add what you wrote here as inline comments in each of the docker-compose files, that would be terrific. These are very clear explanations! 
* For the CI docker file, I think it would be worth pointing out in the code comments that CI/CD in this context does deploys and has sensitive credentials — just to underline that in this context CI has prod access.